bin/prettier.js

```diff
@@ -232,6 +232,21 @@ if (stdin) {
 
     let output;
 
+    if (argv["list-different"]) {
+      try {
+        if (!prettier.check(input, options)) {
+          console.log(filename);
+          process.exitCode = 1;
+        }
+      } catch (e) {
+        // Add newline to split errors from filename line.
+        process.stdout.write("\n");
+        handleError(filename, e);
```

Comment:
Actually, looks like the existing implementation does it incorrectly, good thing you are looking into it :)